### DISCUSSION TITLE
Autolink breaking when a link is posted with a special character

### DISCUSSION THREAD

**Discussion by grzegorzkibitz** [thephpleague/commonmark#1014] [Q&A]
Hi all!

Hope you all are well. I've run into one issue with links which have a special character like `*` in them. Is there a way for the package to recognise the whole URL?

Here is an example URL that breaks: `http://mark.down.com/?something=HJjniUJ87gv*JHgGoHGG`

![image](https://github.com/thephpleague/commonmark/assets/64327566/92ab9705-eb43-42c5-96f6-724c5e51eaf9)
![image](https://github.com/thephpleague/commonmark/assets/64327566/d5026bda-1b28-4dc7-a985-ede32447369f)

Thank you!

**Comment by colinodell** [comment]
Hmm when I test that URL on Babelmark3 I see that this library seems to handle it properly: https://babelmark.github.io/?text=http%3A%2F%2Fmark.down.com%2F%3Fsomething%3DHJjniUJ87gv*JHgGoHGG

Are you sure there's not an extra space in the input?

![Screenshot_20240224-192847.png](https://github.com/thephpleague/commonmark/assets/202034/6e92cdf1-78c2-46f1-b3ca-90de1e72d36a)

**Reply by grzegorzkibitz** [reply]
Completely forgot about this! Nope. There are no spaces there!
